### DISCUSSION TITLE
Source file members read only

### DISCUSSION THREAD

**Discussion by lucat73** [codefori/vscode-ibmi#1223] [Q&A] [answered]
After upgrade "Code for IBM i" to version 1.7.12 when I open a source file in the editor I can't modify it, in the editor tab is displayed a lock.

The previous version 1.7.7 worked flawlessy, after upgrading to VSC version 1.77.1 the object editor became empty and to repopulate it I had to update "Code for IBM i" to actual version.

**Comment by worksofliam** [comment]
So every member you open is locked? Can you ensure that the filter you have created is not protected and also check that connection is not in 'readonly mode' in the connection settings.

**Reply by lucat73** [reply]
Hi Barry, yes every member.

Neither the connection nor the filters are in 'readonly mode'. These settings wasn't changed and up until the update worked correctly.

**Reply by lucat73** [reply, marked as answer]
Nevermind, it seems that saving connection settings although without modifications do the trick.

Thanks for support.
